unstable/lens/struct-struct.rkt

```diff
@@ -0,0 +1,45 @@
+#lang sweet-exp racket/base
+
+provide struct->struct-lens
+
+require racket/local
+        unstable/lens/isomorphism/base
+        for-syntax racket/base
+                   syntax/parse
+                   "private/struct-id.rkt"
+module+ test
+  require lens/private/base/base
+          rackunit
+
+(define-syntax struct->struct-lens
+  (lambda (stx)
+    (syntax-parse stx
+      [(struct->struct-lens s1:struct-id s2:struct-id)
+       (unless (= (length (syntax->list #'[s1.accessor-id ...]))
+                  (length (syntax->list #'[s2.accessor-id ...])))
+         (raise-syntax-error #f (format "~a and ~a have different numbers of fields"
+                                        (syntax-e #'s1) (syntax-e #'s2))
+                             stx))
```

Comment:
I'd like if this error checking was extracted into a helper function. I find it easier to understand macros when verification code like this is separate from the pattern definitions. Maybe something like `(check-struct-field-count-equal #'s1 #'s2)`.